core/src/main/java/jenkins/I18n.java

```diff
@@ -104,10 +113,72 @@ public HttpResponse doResourceBundle(StaplerRequest request) {
             } else if (language != null) {
                 locale = new Locale(language);
             }
-
-            return HttpResponses.okJSON(ResourceBundleUtil.getBundle(baseName, locale));
+            
+            JSONObject json = getBundle(baseName, locale);
+            
+            if (json != null) {
+                return HttpResponses.okJSON(json);
+            }
+            
+            return HttpResponses.error(HttpServletResponse.SC_NOT_FOUND, "No resource bundle found.");
         } catch (Exception e) {
             return HttpResponses.errorJSON(e.getMessage());
         }
     }
+    
+    /**
+     * Get a resource bundle from jenkins or a plugin
+     * @throws MissingResourceException when no bundle is found
+     */
+    @Nonnull
+    public static JSONObject getBundle(String baseName) throws MissingResourceException {
```

Comment:
IMHO the new methods should be Restricted as well.
Otherwise it cannot be considered for backporting